src/UI/Button.elm

```diff
@@ -1,342 +1,427 @@
-        }
+    buttonAny (ClickHref url)
 
 
-withDisabledMode : Bool -> Button msg -> Button msg
-withDisabledMode isDisabled (Button options) =
-    Button
-        { options
-            | isDisabled = isDisabled
-        }
 
+-- Options
 
-disabledColorScheme : ColorScheme
-disabledColorScheme =
-    let
-        default =
-            { bg = Theme.gray4
-            , text = Theme.gray2
-            , border = Theme.gray2
-            }
-    in
-    { normal = default
-    , hover = default
-    }
 
+withMode : ButtonMode -> Button msg -> Button msg
+withMode mode (Button prop opt) =
+    Button prop { opt | mode = mode }
 
-withIcon : Element msg -> Button msg -> Button msg
-withIcon content (Button options) =
-    Button
-        { options
-            | content = Custom content
-        }
-
-
-withPrimaryColor : Button msg -> Button msg
-withPrimaryColor (Button options) =
-    Button
-        { options
-            | colorScheme =
-                { normal =
-                    { bg = Theme.primary
-                    , text = Theme.white
-                    , border = Theme.primary
-                    }
-                , hover =
-                    { bg = Theme.white
-                    , text = Theme.primary
-                    , border = Theme.primary
-                    }
-                }
-        }
-
-
-withSuccessColor : Button msg -> Button msg
-withSuccessColor (Button options) =
-    Button
-        { options
-            | colorScheme =
-                { normal =
-                    { bg = Theme.success
-                    , text = Theme.black
-                    , border = Theme.success
-                    }
-                , hover =
-                    { bg = Theme.white
-                    , text = Theme.black
-                    , border = Theme.success
-                    }
-                }
-        }
-
-
-withDangerColor : Button msg -> Button msg
-withDangerColor (Button options) =
-    Button
-        { options
-            | colorScheme =
-                { normal =
-                    { bg = Theme.error
-                    , text = Theme.white
-                    , border = Theme.error
-                    }
-                , hover =
-                    { bg = Theme.white
-                    , text = Theme.error
-                    , border = Theme.error
-                    }
-                }
-        }
-
-
-withWarningColor : Button msg -> Button msg
-withWarningColor (Button options) =
-    Button
-        { options
-            | colorScheme =
-                { normal =
-                    { bg = Theme.warning
-                    , text = Theme.black
-                    , border = Theme.warning
-                    }
-                , hover =
-                    { bg = Theme.white
-                    , text = Theme.black
-                    , border = Theme.warning
-                    }
-                }
-        }
 
+withStyle : ButtonStyle -> Button msg -> Button msg
```

Comment:
I would kill this, as the user should not be able to set an outline button, at least now :D